### PR TITLE
Update the framer motion dependency to the latest version

### PR DESCRIPTION
**Pull request by youknowriad**
### <samp>🤖 Generated by Copilot at 298e4b6</samp>

Updated `framer-motion` dependency in `packages/components/package.json` to fix animation issues and use new features. Part of a larger dependency update and bug fix pull request.

### <samp>🤖 Generated by Copilot at 298e4b6</samp>

> _`framer-motion` changed_
> _animation bugs are fixed_
> _autumn of updates_

### WHY
Latest is the greatest

### <samp>🤖 Generated by Copilot at 298e4b6</samp>

* Update `framer-motion` dependency to fix animation issues and use latest features ([link](https://github.com/WordPress/gutenberg/pull/49822/files?diff=unified&w=0#diff-2c76dc06185f93bec73660e15338433b95eac6707317564c0f778f5b9abe446cL67-R67))

### CODE REVIEW

**Review by mtias — APPROVED** [review]

**Comment by youknowriad** [comment]
Looks like the package upgrade didn't go as expected. Let me check what happened here.

**Review by mtias — CHANGES_REQUESTED** [review]
I'm seeing type issues on the popover component

**Review by mtias — APPROVED** [review]

**Comment by youknowriad** [comment]
Looks like there's still some JS unit test failures due to the update in the Popover component. Any idea @ciampo

**Comment by youknowriad** [comment]
There's a bunch of. `Warning: An update to %s inside a test was not wrapped in act(...).` issues too. I believe @tyxla or @jsnajdr might some experience with this kind of errors in React upgrade related PRs?

**Comment by jsnajdr** [comment]
It was the `onAnimationComplete` handler that sets the `hasAnimatedOnce` state. It runs in a promise microtask, after render, and is outside `act()`. We've seen this before, and discussed with @ciampo that the `hasAnimatedOnce` check is not needed, because the animation runs only on mount, and is never restarted on any prop change.

File: packages/block-editor/src/components/url-popover/test/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -5,7 +5,7 @@ exports[`URLPopover matches the snapshot in its default state 1`] = `
   <span>
     <div
       class="components-popover block-editor-url-popover is-positioned"
-      style="position: absolute; top: 0px; left: 0px; transform: none;"
+      style="position: absolute; top: 0px; left: 0px; transform: translateZ(0);"
```
Comment (youknowriad) [inline packages/block-editor/src/components/url-popover/test/__snapshots__/index.js.snap:8]: I'm guessing this is an internal implementation details to framer motion animations change. Maybe perf related

**Comment by youknowriad** [comment]
Thanks @jsnajdr

**Comment by ciampo** [comment]
> We've seen this before, and discussed with @ciampo that the `hasAnimatedOnce` check is not needed, because the animation runs only on mount, and is never restarted on any prop change.

Relevant conversation for more context https://github.com/WordPress/gutenberg/pull/46227#discussion_r1037901705

File: packages/components/src/popover/index.tsx [hunk 2]
```diff
@@ -161,7 +148,7 @@ const UnforwardedPopover = (
 		// To avoid overlaps between the standard HTML attributes and the props
 		// expected by `framer-motion`, omit all framer motion props from popover
 		// props (except for `animate`, which is re-defined in `PopoverProps`).
```
Comment (ciampo) [inline packages/components/src/popover/index.tsx]:
```suggestion
		// props (except for `animate` and `children`, which are re-defined in `PopoverProps`).
```

**Comment by youknowriad** [comment]
It looks something in this PR broke the e2e test, in my debugging it seems that when clicking "browse all templates" in the site editor title popover (the one you trigger from the header of the editor), the navigation happens properly but the popover lingers in the DOM for some reason even though it should unmount entirely since it's not rendered anymore.

**Comment by youknowriad** [comment]
I'm also not able to reproduce the popover lingering manually, I only see it in the tests debug trace.

File: packages/block-editor/src/components/url-popover/test/__snapshots__/index.js.snap (hunk 1 above)
Comment (mtias) [reply packages/block-editor/src/components/url-popover/test/__snapshots__/index.js.snap:8]: seems like it

**Comment by mtias** [comment]
Is that e2e test the remaining blocker?

**Comment by youknowriad** [comment]
@mtias yes

**Pull request merged**